tests/x509/test_verification.py

```
@@ -102,3 +157,14 @@ def test_verify_basic():
     chain = verify(ee, policy, [intermediate], store)
 
     assert chain == [ee, intermediate, root]
+
+
+def test_limbo():
+    limbo_file = cryptography_vectors.open_vector_file(
+        os.path.join("x509", "limbo.json"), "r"
+    )
+    with limbo_file:
+        limbo = json.load(limbo_file)
+        testcases = limbo["testcases"]
+        for testcase in testcases:
+            _limbo_testcase(testcase)
```

I _think_ we want to use the `subtests` fixture here, with `subtests.test()` as a context manager to provide distinct subtests for each of these testcases.

Example from another unit test:

https://github.com/trail-of-forks/cryptography/blob/0d213f339146212f517e293f3cf0443931a2701e/tests/x509/test_name.py#L18-L27

---

Ah, I was looking for something like that. Will do that now.

---

This doesn't seem to work (at least now how I was expecting). Even in the `test_name.py` example, if I place an `assert False` within the subtest, I only see one test failure.

---

Yeah, that's surprising (although I admittedly don't know subtests too well). Maybe @reaperhulk or @alex has ideas here 🙂

---

Hahahha, funny story: So the behavior of the upstream `pytest-subtests` does kind of what you'd expect: runs them all.

We have our own custom version that only runs one, and this is because of performance (not the performance of running the tests, but rather the overhead of the subtest machinery itself): https://github.com/pyca/cryptography/blob/main/tests/conftest.py#L52

---

Ah, gotcha -- so your custom version fails fast on the first failure, and only runs them all if they all pass?

---

Makes sense, thanks for explaining!